frame/utility/src/lib.rs

```diff
@@ -184,7 +184,7 @@ decl_module! {
 		///
 		/// The dispatch origin for this call must be _Signed_.
 		#[weight = (
-			T::WeightInfo::as_derivative()
+			T::WeightInfo::as_derivative((*index).into())
```

Comment:
I think as_derivative is `O(1)`, index is not a linear cost (it is only encoded and hashed). But anyway fine.